lib/cocoapods-core/podfile/target_definition.rb

```diff
@@ -663,15 +663,13 @@ def store_pod(name, *requirements)
       # @return [void]
       #
       def store_podspec(options = nil)
-        if options
-          unless options.keys.all? { |key| [:name, :path].include?(key) }
-            raise StandardError, 'Unrecognized options for the podspec ' \
-              "method `#{options}`"
-          end
-          get_hash_value('podspecs', []) << options
-        else
-          get_hash_value('podspecs', []) << { :autodetect => true }
+        options ||= {}
+        unless options.keys.all? { |key| [:name, :path, :subspecs, :subspec].include?(key) }
```

Comment:
I think its worth adding only the plural `:subspecs` option to keep the implementation simpler. If you only want 1 subspec, that could just be `:subspecs => ['Subspec']`

Reply:
I think about it. But I found that I often only use one subspec in a target, so the `subspec=>'xx'` is very simple and very common.